### PR TITLE
Add 2x image url to user badge

### LINKED ISSUE #10434: `UserBadge` from api doesn't return `@2x` image link string
https://osu.ppy.sh/docs/index.html#usercompact-userbadge

Similar to https://github.com/ppy/osu-web/issues/9816.

### PR DESCRIPTION
**Pull request by kj415j45**
Resolve #10434

~~I'm not sure if naming the new field as `image@2x` and duplicating `image_url` as `image` would be better or not.~~

### CODE REVIEW

File: app/Transformers/UserBadgeTransformer.php [hunk 1]
```diff
@@ -11,10 +11,14 @@ class UserBadgeTransformer extends TransformerAbstract
 {
     public function transform(UserBadge $badge)
     {
+        $image_url = $badge->imageUrl();
+        $image2x_url = retinaify($image_url);
```
Comment (nanaya) [inline app/Transformers/UserBadgeTransformer.php:15]: camel case variable name

File: resources/js/interfaces/user-badge-json.ts [hunk 2]
```diff
@@ -5,5 +5,6 @@ export default interface UserBadgeJson {
   awarded_at: string;
   description: string;
   image_url: string;
+  'image@2x_url': string;
```
Comment (nanaya) [inline resources/js/interfaces/user-badge-json.ts:8]: as per lint check, this isn't sorted correctly

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — CHANGES_REQUESTED** [review]
the code doesn't work

**Comment by kj415j45** [comment]
🤦 I should use rename symbol next time.

**Review by nanaya — CHANGES_REQUESTED** [review]
just noticed the docs in `user_compact.md` needs an update as well

**Review by nanaya — APPROVED** [review]

**Pull request merged**